### DISCUSSION TITLE
Python Export Chaser with Args

### DISCUSSION THREAD

**Discussion by scott-wilson** [Autodesk/maya-usd#2001] [Q&A] [answered]
Hey all,

I have some example Python code like this (using 0.13 of the Maya USD):

```python
import mayaUsd.lib
from pxr import Usd


class PublishAssetExportChaser(mayaUsd.lib.ExportChaser):
    def ExportDefault(self):
        print("EXPORT DEFAULT", dir(self))

        return self.ExportFrame(Usd.TimeCode.Default())

    def ExportFrame(self, frame):
        print("EXPORT FRAME", frame, dir(self))

        return True

    def PostExport(self):
        print("POST EXPORT", dir(self))

        return True


mayaUsd.lib.ExportChaser.Register(PublishAssetExportChaser, "publish_asset")
```

And I want to run the following export command with chaser args:

```python
cmds.mayaUSDExport(..., chaser=["publish_asset"], chaserArgs=[("publish_asset", "myarg", 1)])
```

How would I pass that arg into the class? Or, would it just be better to have a post process that opens the USD scene and makes the required change? From what I understand, the nice thing about the chasers is that I already have the USD scene open and can make changes there. (In my case, I want to change the path to texture files on publish.)

**Comment by kxl-adsk** [comment, marked as answer]
Hi Scott. I believe we fixed this in the following PR https://github.com/Autodesk/maya-usd/pull/1864 which is part of the 0.15.0 version.

Cheers,
KxL
